src/util/unicode.cpp

```diff
@@ -199,10 +199,24 @@ std::wstring utf8_to_utf16_native_endian(const std::string &in)
 {
   std::wstring result;
   result.reserve(in.size());
+
+  for(auto codepoint : utf8_to_utf32(in))
+    utf16_append_code(codepoint, result);
+
+  return result;
+}
+
+/// Convert UTF8-encoded string to UTF-32 with architecture-native endianness.
+/// \par parameters: String in UTF-8 format
+/// \return String in UTF-32 format.
+std::basic_string<unsigned int> utf8_to_utf32(const std::string &utf8_str)
```

Comment:
I'd recommend [`char32_t`](https://en.cppreference.com/w/cpp/keyword/char32_t) instead of `unsigned int`, because it is specifically meant for this purpose.

Reply:
Ok, now using `std::u32string`